updating/understanding-openshift-upgrade-duration.adoc

```diff
@@ -0,0 +1,91 @@
+The Machine Config Operator (MCO) applies a new machine configuration to each control plane and worker node. During this process, the MCO performs the following tasks on each node of the cluster:
+
+* Node draining
+* OS update staging
+* Node rebooting
+
+The control plane node upgrade duration is more predictable and oftentimes shorter than worker nodes since it does not include large workloads. You can upgrade the worker nodes in parallel by setting the `maxUnavailable` field to greater than 1 in the machine configuration pool. The MCO cordons the number of nodes specified in `maxUnavailable` and marks them unavailable for upgrade.
+
+[IMPORTANT]
+====
+With {product-title} 4 upgrades, an operating system lifecycle management and maintenance is made as a part of the cluster upgrade.  This only applies to {op-system-first} nodes.
+====
+
+The time to complete this process depends on many factors including the node and infrastructure configuration. This process might take 5 minutes to complete per node.
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../architecture/architecture-installation.html#update-service-overview_architecture-installation[About the OpenShift Update Service]
+
+
+== Factors affecting the upgrade duration
+In addition to Cluster Version Operator (CVO) and Machine Config Operator (MCO), the following factors also impact the upgrade duration:
+
+* Pod Disruption Budget (PDB)
+* Cluster node health
+* Number of nodes in the cluster
+* Red Hat Enterprise Linux (RHEL) nodes
+
+=== Pod disruption budget
+You must consider pod disruption budgets and pod anti-affinity when you configure `maxUnavailable` field in the machine configuration pool.
+
+For example, if there is a cluster with three worker nodes, you must configure the disruption budget such that it has minimum two replicas be always running. When you configure pod anti-affinity, you must specify that two replicas should never run on the same node. To avoid pod scheduling issues, you must ensure the availability of minimum two worker nodes.
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../nodes/pods/nodes-pods-configuring.adoc#nodes-pods-configuring-pod-distruption-about[pod disruption budget]
+* xref:../nodes/scheduling/nodes-scheduler-pod-affinity.adoc#nodes-scheduler-pod-affinity-about[pod anti-affinity]
+
+=== Cluster node health
+An unhealthy node takes longer time to upgrade. It is recommended to resolve any issues before you begin the cluster upgrade.
+
+=== Red Hat Enterprise Linux nodes
+{op-system-base-full} worker nodes require an additional usage of Openshift-ansible to update node binary components. This rolling update is similar to the MCO initiated rolling update and therefore it has a similar duration. The time it takes to reboot a particular node varies significantly. In cloud instances, the reboot takes less than a minute whereas in physical bare metal hosts the reboot takes more than 15 minutes.
```

Comment:
"Openshift" -> "OpenShift".  And looks like "Ansible" is the canonical casing too.

Reply:
Also maybe "may take less than a minute"?  Because while many cloud instance reboot quickly, I don't think we want to commit to all cloud instances always rebooting quickly.

Reply:
@wking Done.

Reply:
@wking what is the context of Openshift-Ansible here ? Is it [OpenShift Ansible broker](https://docs.openshift.com/container-platform/3.11/architecture/service_catalog/ansible_service_broker.html) ?

Reply:
No, the context is using Ansible to update RHEL compute, as described [here][1].

[1]: https://docs.openshift.com/container-platform/4.10/updating/updating-cluster-rhel-compute.html#updating-cluster-rhel-compute-hooks